addons/mail/models/mail_thread.py

```diff
@@ -1888,6 +1888,9 @@ def _check_can_update_message_content(self, message):
             raise exceptions.UserError(_("Only logged notes can have their content updated on model '%s'", self._name))
         if message.tracking_value_ids:
             raise exceptions.UserError(_("Messages with tracking values cannot be modified"))
+        if not message.message_type == 'comment':
+            raise exceptions.UserError(_("Only messages type comment can have their content updated on model 'mail.channel'"))
```

Comment:
Note that mail_thread is generic. It should not have any reference to mail.channel. Moreover UserError may be displayed to user, so we should avoid technical names.